src/hdmf/backends/io.py

```diff
@@ -188,6 +188,14 @@ def close(self):
         ''' Close this HDMFIO object to further reading/writing'''
         pass
 
+    @staticmethod
+    def generate_dataset_html(dataset):
```

Comment:
@oruebel as we discussed in person, I tested locally that this function was triggered with `ZarrIO` and added a test for the `HDMFIO` method since `ZarrIO` will (hopefully) soon have its own implementation.